advtraj/cli/uclales.py

```diff
@@ -99,14 +104,23 @@ def main(data_path, file_prefix, output_path):
     ds = load_data(files=files, fields_to_keep=["w"])
 
     ds_ = ds.isel(time=int(ds.time.count()) // 2).sel(z=slice(300, None))
-    da_pt = ds_.where(ds_.w == ds_.w.max(), drop=True)
-
-    # n_timesteps = int(ds.time.count())
-    ds_starting_points = xr.Dataset()
-    ds_starting_points["x"] = da_pt.x.item()
-    ds_starting_points["y"] = da_pt.y.item()
-    ds_starting_points["z"] = da_pt.z.item()
-    ds_starting_points["time"] = da_pt.time
+
```

Comment:
Could you explain why it's better to use `np.where(...)` rather than `ds.where(...)` here? In general I think it is better to avoid using bare numpy arrays (and constructing xr.DataArray's anew) where possible. Without running the code I think the main change is that `time` has become a coordinate. I think that could be achieved with [ds.expand_dims(...)](https://xarray.pydata.org/en/stable/generated/xarray.Dataset.expand_dims.html?highlight=expand_dims) by replacing the original line 109

Reply:
Put simply, your code only works for a single trajectory - `.item()`. 
I tried extending this to choose points >= 0.9 w max (i.e.
`da_pt = ds_.where(ds_.w  >= 0.9 * ds_.w.max(), drop=True)`
but this doesn't work (even replacing `.item()` with` .values()`) - you get a dataset retaining all the (x, y, z) points where 
the condition is met for any of w(x, y, z) w.g. if the codition is met at w[1, 2, 3] and w[2, 3, 4] da_pt has all the x[1,2], y[2,3] and z[3,4] points, i.e. 8 not 2. 
xarrays 'where' just doesn't work like numpys, and I haven't figured out how to do what's needed in xarray. 
I'd love someone to show me, of course.